src/libcadet/model/reaction/MichaelisMentenReaction.cpp

```diff
@@ -28,313 +28,584 @@
+        if (paramProvider.exists("MM_STOICHIOMETRY_BULK"))
+        {
+            const std::vector<double> s = paramProvider.getDoubleArray("MM_STOICHIOMETRY_BULK");
+            std::vector<double> KIC(_stoichiometryBulk.columns() * _nComp * _nComp); 
+            std::vector<double> KIUC(_stoichiometryBulk.columns() * _nComp * _nComp);
+            bool hasCompetiveInhibition = false;
+            
+            if (paramProvider.exists("MM_KI_C"))
+            {
+                KIC = paramProvider.getDoubleArray("MM_KI_C");
+                if (KIC.size() != _stoichiometryBulk.columns() * _nComp * _nComp)
+                    throw InvalidParameterException("MM_KI_C must have the size (number of reactions) x (number of components) x (number of components)");
+            }
+
+            if (paramProvider.exists("MM_KI_UC"))
+            {
+                KIUC = paramProvider.getDoubleArray("MM_KI_UC");
+                if (KIUC.size() != _stoichiometryBulk.columns() * _nComp * _nComp)
+                    throw InvalidParameterException("MM_KI_UC must have the size (number of reactions) x (number of components) x (number of components)");
+            }
+            if (paramProvider.exists("MM_KI"))
+            {
+                KIUC = paramProvider.getDoubleArray("MM_KI");
+                KIC = paramProvider.getDoubleArray("MM_KI");
+            }
+
+
+            if (s.size() != _stoichiometryBulk.elements())
+                throw InvalidParameterException("MM_STOICHIOMETRY_BULK size mismatch: Expected " +
+                    std::to_string(_stoichiometryBulk.elements()) + " elements but got " + std::to_string(s.size()));
+
+            std::copy(s.begin(), s.end(), _stoichiometryBulk.data());
+
+            // Identify substrates (negative entries in stoichiometry matrix)
+            const unsigned int nReactions = static_cast<unsigned int>(_stoichiometryBulk.columns());
+            _idxSubstrate.clear();
+            _idxCompInhibitors.resize(nReactions);
+            _idxUncompInhibitors.resize(nReactions);
+
+            for (unsigned int r = 0; r < nReactions; ++r)
+            {
+                std::vector<int> idxSubstrateReaction_r;
+                for (unsigned int c = 0; c < _nComp; ++c)
+                {
+                    if (_stoichiometryBulk.native(c, r) < 0.0)
+                        idxSubstrateReaction_r.push_back(static_cast<int>(c));
+                }
+
+                if (idxSubstrateReaction_r.empty())
+                    throw InvalidParameterException("Michaelis Menten: No substrates found for reaction " + std::to_string(r));
+                if (idxSubstrateReaction_r.size() > 1 && _oldInterface)
+                    throw InvalidParameterException("Michaelis Menten: The old interface is used which does not support multible substrats. Pleas refer to the documentation of CADET version >= 5.2.0");
```

Comment:
```suggestion
                    throw InvalidParameterException("Michaelis Menten: The old interface is used which does not support multiple substrates. Pleas refer to the documentation of CADET version >= 5.2.0");
```

multible -> multiple
substrats -> substrates